### PR TITLE
WIP: verification (#3893)

### PR DESCRIPTION
**Pull request by traumschule**
Current plan is to collapse the page with a [css-only accordeon](https://tympanus.net/codrops/2012/02/21/accordion-with-css3/) and split it in basic steps
1. introduction
2. install gpg
3. download signing key
4. verify gpg signature
5. verify checksums

Each section will have OS specific instructions. When all sections are collapsed at the beginning and users only open what they are interested in, my hope is the page will be much less confusing.

### CODE REVIEW

File: docs/en/verifying-signatures.wml [hunk 1]
```diff
@@ -9,199 +9,380 @@
...
-    <hr>
-    <p>First of all you need to have GnuPG installed before you can verify signatures.
-    Download it from <a href="https://gpg4win.org/download.html">https://gpg4win.org/download.html</a>.</p>
-    <p>Once it's installed, use GnuPG to import the key that signed your
+  </label>
+  <article class="ac-box">
+
+   <p><span class="hint">Choose your operating system:</span>
+   <label for="ac-2-1"><a name="KeyWindows">&#9658; Windows</a></label>
+   <label for="ac-2-2"><a name="KeyMacOS">&#9658; Mac OS X</a></label>
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: Apple calls it "macOS", but it might be ok to use "Mac OS X" if that name is easier for users.

File: docs/en/verifying-signatures.wml [hunk 2]
```diff
@@ -9,199 +9,380 @@
...
-    <p>The Tor Browser team signs Tor Browser releases. Import its key
-    (0x4E2C6E8793298290) by starting <i>cmd.exe</i> and typing:</p>
-    <pre>gpg.exe --keyserver pool.sks-keyservers.net --recv-keys 0x4E2C6E8793298290</pre>
-    <p>After importing the key, you can verify that the fingerprint
-    is correct:</p>
-    <pre>gpg.exe --fingerprint 0x4E2C6E8793298290</pre>
+    </p>
+    <p>
+    The Tor Browser team signs Tor Browser releases. Import its key
+    (0x4E2C6E8793298290) by starting <i>cmd.exe</i> and typing:
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: Always use long key ids. Short key ids are easy to fake.

File: docs/en/verifying-signatures.wml [hunk 3]
```diff
@@ -9,199 +9,380 @@
...
+    <h3>Import OpenPGP key on Mac OS</h3>
+    <p>
+    You need to have GnuPG installed before you can verify
+    signatures. Install it from
+    <a href="https://www.gpgtools.org/">https://www.gpgtools.org/</a>.
+    </p>
+
+    <p>The next step is to use GnuPG to import the key that signed
+    your package. The Tor Browser team signs Tor Browser releases. Import its
+    key (0x4E2C6E8793298290) by starting the terminal under "Applications"
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: Terminal.app is in Applications/Utilities on macOS

File: docs/en/verifying-signatures.wml [hunk 4]
```diff
@@ -9,199 +9,380 @@
...
+   </article>
+  </div>
+<!-- Linux -->
+  <div>
+   <input id="ac-2-3" name="accordion-2" type="radio" />
+   <article class="ac-os">
+    <h3>Import OpenPGP key on Linux</h3>
+    <p>
+    You need to have GnuPG installed before you can verify
+    signatures. It's probably GnuPG is alreadyy installed on your
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: Typo: alreadyy

File: docs/en/verifying-signatures.wml [hunk 5]
```diff
@@ -9,199 +9,380 @@
...
+    The next step is to use GnuPG to import the key that signed
+    your package. The Tor Browser team signs Tor Browser releases. Import its
+    key (0x4E2C6E8793298290) by starting the terminal and typing:</p>
+    <pre>
+    $ gpg --keyserver pool.sks-keyservers.net --recv-keys 0x4E2C6E8793298290
+    </pre>
+    <p>
+   </article>
+  </div>
+  <div>
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: It is unclear if this div is for Linux or all OSes. Please add a comment.

Comment (teor2345) [reply docs/en/verifying-signatures.wml]: (It should be for all OSes.)

File: docs/en/verifying-signatures.wml [hunk 6]
```diff
@@ -9,199 +9,380 @@
...
-    <p>You should see:</p>
     <pre>
-pub   rsa4096/0x4E2C6E8793298290 2014-12-15 [C] [expires: 2020-08-24]
-      Key fingerprint = EF6E 286D DA85 EA2A 4BA7  DE68 4E2C 6E87 9329 8290
-uid                   [ unknown] Tor Browser Developers (signing key) <torbrowser@torproject.org>
-sub   rsa4096/0xD1483FA6C3C07136 2016-08-24 [S] [expires: 2018-08-24]
-      Key fingerprint = A430 0A6B C93C 0877 A445  1486 D148 3FA6 C3C0 7136
-sub   rsa4096/0xEB774491D9FF06E2 2018-05-26 [S] [expires: 2020-09-12]
-      Key fingerprint = 1107 75B5 D101 FB36 BC6C  911B EB77 4491 D9FF 06E2
+gpg: assuming signed data in 'tor-browser-osx64-<version-torbrowserbundleosx64>_en-US.tar.xz'
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: This is the Linux file name

File: docs/en/verifying-signatures.wml [hunk 7]
```diff
@@ -9,199 +9,380 @@
...
+    Currently valid subkey fingerprints are:
+    </p>
     <pre>
     5242 013F 02AF C851 B1C7  36B8 7017 ADCE F65C 2036
     BA1E E421 BBB4 5263 180E  1FC7 2E1A C68E D408 14E0
-    A430 0A6B C93C 0877 A445  1486 D148 3FA6 C3C0 7136</pre></p>
+    A430 0A6B C93C 0877 A445  1486 D148 3FA6 C3C0 7136
+    </pre>
     <p>
     Notice that there is a warning because you haven't assigned a trust
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: Is this section in the right place in the document?

Comment (traumschule) [reply docs/en/verifying-signatures.wml]: it is for all OS. for linux it might be confusing though, moving the debian note up.

File: docs/en/verifying-signatures.wml [hunk 8]
```diff
@@ -210,113 +391,250 @@ Primary key fingerprint: EF6E 286D DA85 EA2A 4BA7  DE68 4E2C 6E87 9329 8290
     exchange key fingerprints.
```
[the remark this answers is not in the document]
Comment (traumschule) [reply docs/en/verifying-signatures.wml]: Good point, changing to "As international travel to meet the developer might be
    unfeasable you are left with trusting other people who signed this key."

File: docs/en/verifying-signatures.wml (hunk 2 above)
Comment (traumschule) [reply docs/en/verifying-signatures.wml]: Isn't this the long id? It's the same as listed on our [signing-keys](https://www.torproject.org/docs/signing-keys.html.en) page.
```
$ gpg2 --keyid-format long --list-key 0x4E2C6E8793298290
pub   rsa4096/4E2C6E8793298290 2014-12-15 [C] [expires: 2020-08-24]
      Key fingerprint = EF6E 286D DA85 EA2A 4BA7  DE68 4E2C 6E87 9329 8290
uid                 [ unknown] Tor Browser Developers (signing key) <torbrowser@torproject.org>
sub   rsa4096/EB774491D9FF06E2 2018-05-26 [S] [expires: 2020-09-12]
```

File: docs/en/verifying-signatures.wml [hunk 9]
```diff
@@ -210,113 +391,250 @@ Primary key fingerprint: EF6E 286D DA85 EA2A 4BA7  DE68 4E2C 6E87 9329 8290
...
+    <label for="ac-4-3"><a name="ChecksumsLinux">&#9658; Linux</a></label>
+   </span>
+   </p>
+
+<!-- Windows -->
+  <div>
+    <input id="ac-4-1" name="accordion-4" type="radio" checked />
+   <article class="ac-os">
+    <!--<pre id="ttb-key">
+    > gpg.exe --keyserver pool.sks-keyservers.net --recv-keys 0x4E2C6E8793298290
```
[the remark this answers is not in the document]
Comment (traumschule) [reply docs/en/verifying-signatures.wml]: This is commented because I assume at this point it is ok, to tell users reproduce above step to import the key by just telling the key id. I see no better option at the moment.

File: docs/en/verifying-signatures.wml [hunk 10]
```diff
@@ -9,199 +9,380 @@
...
-    <p>The next step is to use GnuPG to import the key that signed
-    your package. The Tor Browser team signs Tor Browser releases. Import its
-    key (0x4E2C6E8793298290) by starting the terminal (under "Applications"
-    in Mac OS X) and typing:</p>
-
-    <pre>gpg --keyserver pool.sks-keyservers.net --recv-keys 0x4E2C6E8793298290</pre>
-
-    <p>After importing the key, you can verify that the fingerprint
-    is correct:</p>
+    <pre>
```
[the remark this answers is not in the document]
Comment (traumschule) [reply docs/en/verifying-signatures.wml]: Maybe command-O needs to be explained? I changed it to:
```
    <p>
    The easiest way to verify an "asc" file is to open it by double-clicking
    on it, or using the keying command-O. Alternatively type into the terminal:
    </p>

    <pre>
    $ gpg --verify ~/Downloads/TorBrowser-<version-torbrowserbundleosx64>_en-US.dmg{.asc*,}
    </pre>
```

File: docs/en/verifying-signatures.wml (hunk 2 above)
Comment (teor2345) [reply docs/en/verifying-signatures.wml]: Oh, ok, I usually use the full key id. But if that's what we use, that's ok.

File: docs/en/verifying-signatures.wml (hunk 10 above)
[the remark this answers is not in the document]
Comment (teor2345) [reply docs/en/verifying-signatures.wml]: I think "typing" is probably better than "using the keying".

File: docs/en/verifying-signatures.wml [hunk 11]
```diff
@@ -387,24 +393,25 @@ Primary key fingerprint: EF6E 286D DA85 EA2A 4BA7  DE68 4E2C 6E87 9329 8290
     Notice that there is a warning because you haven't assigned a trust
     index to this person. This means that GnuPG verified that the key made
     that signature, but it's up to you to decide if that key really belongs
-    to the developer. The best method is to meet the developer in person and
-    exchange key fingerprints.
+    to the developer. As international travel to meet the developer might be
+    unfeasable you are left with trusting other people who signed this key.
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: This explanation raises security concerns, but it doesn't tell people what they should do.

If you can't give people a useful action to take to improve their security, please just tell them to ignore the warning.

One useful action might be: "use another device or another internet connection to check the key fingerprints listed on the tor website at ..."

File: docs/en/verifying-signatures.wml [hunk 12]
```diff
@@ -507,7 +516,7 @@ Primary key fingerprint: EF6E 286D DA85 EA2A 4BA7  DE68 4E2C 6E87 9329 8290
 <!-- Mac OS --><!--
  <div>
    <article class="ac-os">
-     This process does not work on OS X yet due to Apple's codesigning requirement.
+     This process does not work on macOS yet due to Apple's codesigning requirement.
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: Please consistently use "OS X", "macOS", or "Mac OS X" throughout the document.

You could choose "macOS", because that's what Apple calls it now. Or you could match the Tor download page.

File: docs/en/verifying-signatures.wml (hunk 9 above)
[the remark this answers is not in the document]
Comment (teor2345) [reply docs/en/verifying-signatures.wml]: You should feel free to delete obsolete instructions.
We can always get them back out of git.

File: docs/en/verifying-signatures.wml [hunk 13]
```diff
@@ -9,314 +9,641 @@
...
+   </article>
+ </div>
+-->
+
+<!-- Linux -->
+  <div>
+   <input id="ac-4-3" name="accordion-4" type="radio" />
+   <article class="ac-os ac-4-3">
+    <!--<pre id="ttb-key">
```
Comment (teor2345) [inline docs/en/verifying-signatures.wml]: You should feel free to delete obsolete instructions.
We can always get them back out of git.

**Comment by hiromipaw** [comment]
Hey, could you just resolve conflicts before I merge this?

**Comment by traumschule** [comment]
mh, the rebase added all commit with new hashes. will you squash it all?

**Comment by traumschule** [comment]
rebased.

@teor2345 sorry, lost your comments through `push -f`. Something above is still unresolved.

> I feel like this document is way too wordy, but that might be ok as a reference for advanced users.

Maybe that's ok because we (hopefully) soon have are shorter version in the tb-manual:
http://ea5faa5po25cf7fb.onion/projects/tor/ticket/9843

**Comment by traumschule** [comment]
rebased on top of #12

**Comment by traumschule** [comment]
squashed commits to `docs/en/verification.wml` into one and removed unrelated changes to make reviewing easier.

**Comment by traumschule** [comment]
updated css.
@teor2345 are you still requesting changes here?
